### ISSUE TITLE
[Offline] offline will never supported??

### ISSUE THREAD

**Issue by shinriyo** [CartoDB/mobile-sdk#539]
[offline](https://carto.com/developers/mobile-sdk/support/contribute/)

It says `This component is still under support but it will not be further developed. We don’t recommend starting new projects with it.`

so do you have any alternative proposal?

**Comment by shinriyo** [comment]
Thank you for replying.
I tried to use the sample.
https://github.com/CartoDB/mobile-ios-samples

I registered my account but I can't find `License` key.

I want to replace it.
https://github.com/CartoDB/mobile-ios-samples/blob/master/HelloMap.Swift/HelloMap.Swift/AppDelegate.swift#L17

**Comment by farfromrefug** [comment]
@shinriyo i am not part of carto so i cant say about the license part. I would use it without any carto stuff. What you are planning on doing with it? which map data are you planning to use? 
I should point out that you should differentiate the SDK from carto as a map data provider. Here i am talking about the SDK. 
If i was to select a map data provider i would go for maptiler https://docs.maptiler.com/cloud/api (i personally use offline map data i generate or openstreetmap)

**Comment by shinriyo** [comment]
Thank you for your response.
One feature I desire is a map of Japan.
Once downloaded, I would like it to be guaranteed to be retained offline for at least a month.

Also, rather than the current settings on iOS, I want to switch the language displayed on the map to Japanese, English, Simplified Chinese, Traditional Chinese, and Korean by passing parameters.

The ability to zoom in, zoom out, and change the coordinates position is also necessary.

**Comment by shinriyo** [comment]
@farfromrefug 
Is the server a web server?
Can I store data on my iPad?

**Comment by farfromrefug** [comment]
@shinriyo yes the server would be a web server you setup. The downloaded files would be a mbtiles and it could be stored on an ipad for as long as you want. Though be careful a map of japan might be quite a big file.
Here is how you can generate such a file https://github.com/onthegomap/planetiler

**Comment by shinriyo** [comment]
@farfromrefug 
I don't understand how to combine them.
So, you're saying we use CartoMobileSDK to manipulate the map, use maptiler for the map data itself, and use planetiler as the server to place data on the map?

**Comment by farfromrefug** [comment]
No planetiler can generate map data which either can be used offline or served through a web server. Maptiler is simply a provider of map tiles so you dont have to generate anything.

**Comment by shinriyo** [comment]
@farfromrefug 

I simply want to locally place a map of Japan (just Western Japan is fine) on an iPad, and just display, zoom, and support multiple languages on the map. However, I lack the knowledge related to maps, and it's too difficult for me to understand.

I tried using this sample, but it requires data called "mbtiles". I'm unsure about how to obtain this data. Do I need Docker?
I just want to download a map of Japan with a click of a button.

**Comment by shinriyo** [comment]
I don't understand why planetiler is necessary in the first place.
Is it used to generate .osm.pbf files?
Does this file also include language information?
If I have the .osm.pbf, do I no longer need the mbtiles?
I'm not specialized in mapping, so I don't understand.

What I want to do is just the following:

- Download a map of western Japan.
- Zoom in.
- Focus on the desired coordinates.
- Map language support (English, Japanese, Traditional Chinese, Simplified Chinese, Korean).

**Comment by lionpham-gis** [comment]
> I would say use it. It is 100% usable and some fork might come out hopefully soon enough. I am already maintaining my fork but we might start an organized effort

That's great, can you build it into an aar file or publish it to maven?

**Comment by farfromrefug** [comment]
@lionpham-gis yes I could. But I would prefer for this to get organized . the current published version is really stable. No major bug to fix so I think we can wait. Just know that it is not abandoned those currently the Dev is pretty slow.

**Comment by jaakla** [comment]
> I don't understand why planetiler is necessary in the first place.
> Is it used to generate .osm.pbf files?

.osm.pbf is (usually) source OSM data what is just technically not usable for rendering map. These processing tools convert the data to mbtiles which is basically the same data (even uses also same pbf as encoding format), but info is reorganized heavily for fast display. The data map is indexed to map tiles and zoom levels, has presimplified data etc. Mbtiles also has selection of languages for labels, so I suggest to check if the processing includes all the languages you need.

**Comment by rexfordnyrk** [comment]
> I don't understand why planetiler is necessary in the first place. Is it used to generate .osm.pbf files? Does this file also include language information? If I have the .osm.pbf, do I no longer need the mbtiles? I'm not specialized in mapping, so I don't understand.
> 
> What I want to do is just the following:
> 
> * Download a map of western Japan.
> * Zoom in.
> * Focus on the desired coordinates.
> * Map language support (English, Japanese, Traditional Chinese, Simplified Chinese, Korean).

Were you able to get it done? if not hit me up on @rexfordnyrk twitter
